### PR TITLE
Fix keyboard navigation with hidden anchor tabs

### PR DESCRIPTION
**Pull request by m-akinc**
# Pull Request

## 🤨 Rationale

Part of #1383
The tabs control needs to be fixed in FAST, but we have similar flawed logic being used for the anchor tabs.

## 👩‍💻 Implementation

The code already handles disabled tabs in the keyboard navigation logic. We just need to expand that to handle hidden tabs the same way. I updated `isFocusableElement()` to also look for the presence of `hidden`.

Also, to make Home/End work properly I had to replace a call to `isDisabledElement()` with `isFocusableElement()` (which used to be equivalent, but now are not).

## 🧪 Testing

Wrote two new test cases and tested manually in Storybook.

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: packages/nimble-components/src/anchor-tabs/tests/anchor-tabs.spec.ts [hunk 1]
```diff
@@ -203,6 +204,20 @@ describe('AnchorTabs', () => {
             initialFocusIndex: 0,
             keyName: keyEnd,
             expectedFinalFocusIndex: 1
+        },
+        {
+            name: 'should skip hidden tab when arrowing right',
```
Comment (mollykreis) [inline packages/nimble-components/src/anchor-tabs/tests/anchor-tabs.spec.ts:209]: The test cases for hidden tabs don't seem quite as complete as the test cases for disabled tabs. Is there a reason not to have a test for arrowing left or for pressing the Home key?

File: packages/nimble-components/src/anchor-tabs/index.ts [hunk 2]
```diff
@@ -104,8 +104,12 @@ export class AnchorTabs extends FoundationElement {
         return el.getAttribute('aria-disabled') === 'true';
     };
 
+    private readonly isHiddenElement = (el: Element): el is HTMLElement => {
+        return el.getAttribute('hidden') !== null;
```
Comment (mollykreis) [inline packages/nimble-components/src/anchor-tabs/index.ts:108]: Is there a reason not to use `hasAttribute('hidden')` here instead?

**Review by mollykreis — APPROVED** [review]

File: packages/nimble-components/src/anchor-tabs/tests/anchor-tabs.spec.ts (hunk 1 above)
Comment (m-akinc) [reply packages/nimble-components/src/anchor-tabs/tests/anchor-tabs.spec.ts:209]: I know they exercise the same relevant code paths, so I didn't write them. But then the same logic would apply to disabled tabs... so I guess I should add more test cases for consistency.

**Review by jattasNI — APPROVED** [review]
I think you mentioned you'd filed an issue/created a PR in FAST. If so can you make sure it's linked from the Nimble issue?

**Review by rajsite — APPROVED** [review]

**Pull request merged**